### PR TITLE
get recommended and similar movies

### PR DESCRIPTION
**Pull request by bariscanyilmaz**
Added [Recommendations](https://developers.themoviedb.org/3/movies/get-movie-recommendations) and [Similar](https://developers.themoviedb.org/3/movies/get-similar-movies) movies feature

### CODE REVIEW

**Comment by nCubed** [comment]
Thanks for the PR. I've been a bit backed up, but will review in detail in the next couple of days. Cheers!

File: DM.MovieApi.IntegrationTests/MovieDb/Movies/ApiMovieRequestTests.cs [hunk 1]
```diff
@@ -350,5 +350,7 @@ public async Task GetPopularAsync_CanPageResults()
             await ApiResponseUtil.AssertCanPageSearchResponse( "unused", minimumPageCount, minimumTotalResultsCount,
                 ( _, page ) => _api.GetPopularAsync( page ), x => x.Id );
         }
+
```
Comment (nCubed) [inline DM.MovieApi.IntegrationTests/MovieDb/Movies/ApiMovieRequestTests.cs:353]: Undo the changes to this file as the only change are extra lines.

File: DM.MovieApi.IntegrationTests/MovieDb/Movies/ApiMovieRequestTests_GetRecommendations.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
+using System.Threading.Tasks;
+using DM.MovieApi.ApiResponse;
+using DM.MovieApi.MovieDb.Movies;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace DM.MovieApi.IntegrationTests.MovieDb.Movies
+{
+
```
Comment (nCubed) [inline DM.MovieApi.IntegrationTests/MovieDb/Movies/ApiMovieRequestTests_GetRecommendations.cs:8]: Remove extra lines in file.

File: DM.MovieApi.IntegrationTests/MovieDb/Movies/ApiMovieRequestTests_GetRecommendations.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    [TestClass]
+    public class GetRecommendationsTests
+    {
+        private IApiMovieRequest _api;
+
+        [TestInitialize]
+        public void TestInit()
+        {
+
```
Comment (nCubed) [inline DM.MovieApi.IntegrationTests/MovieDb/Movies/ApiMovieRequestTests_GetRecommendations.cs:17]: Remove extra lines in methods.

File: DM.MovieApi.IntegrationTests/MovieDb/Movies/ApiMovieRequestTests_GetRecommendations.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        public async Task GetRecommendationsAsync_HasError_InvalidMovieId()
+        {
+            const int movieId = 1;
+
+            ApiSearchResponse<MovieInfo> response = await _api.GetRecommendationsAsync(movieId);
+            Assert.IsNotNull(response.Error);
+            
+        }
+    }
+}
```
Comment (nCubed) [inline DM.MovieApi.IntegrationTests/MovieDb/Movies/ApiMovieRequestTests_GetRecommendations.cs:49]: All files should have a newline at the end.

File: DM.MovieApi.IntegrationTests/MovieDb/Movies/ApiMovieRequestTests_GetSimilar.cs [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        public void TestInit()
+        {
+
+            ApiResponseUtil.ThrottleTests();
+            _api = MovieDbFactory.Create<IApiMovieRequest>().Value;
+
+        }
+
+        [TestMethod]
+        public async Task GetSimilarsAsync_Returns_ValidResults()
```
Comment (nCubed) [inline DM.MovieApi.IntegrationTests/MovieDb/Movies/ApiMovieRequestTests_GetSimilar.cs:24]: Fix the spelling error in method name.

File: DM.MovieApi/MovieDb/Movies/ApiMovieRequest.cs [hunk 6]
```diff
@@ -152,7 +152,51 @@ public async Task<ApiQueryResponse<MovieCredit>> GetCreditsAsync( int movieId, s
...
+
+            return response;
+        }
+
+        public async Task<ApiSearchResponse<MovieInfo>> GetRecommendationsAsync(int movieId, int pageNumber = 1, string language = "en")
+        {
+            var param = new Dictionary<string, string>
+            {
+                {"language", language},
+                {"append_to_response", "keywords"},
```
Comment (nCubed) [inline DM.MovieApi/MovieDb/Movies/ApiMovieRequest.cs:165]: If using "append_to_response", then the backing integration tests must test the "keywords" are present in the response.

File: DM.MovieApi/MovieDb/Movies/ApiMovieRequest.cs [hunk 7]
```diff
@@ -152,7 +152,51 @@ public async Task<ApiQueryResponse<MovieCredit>> GetCreditsAsync( int movieId, s
...
+
+            return response;
+        }
+
+        public async Task<ApiSearchResponse<MovieInfo>> GetSimilarAsync(int movieId, int pageNumber = 1, string language = "en")
+        {
+            var param = new Dictionary<string, string>
+            {
+                {"language", language},
+                {"append_to_response", "keywords"},
```
Comment (nCubed) [inline DM.MovieApi/MovieDb/Movies/ApiMovieRequest.cs:187]: If using "append_to_response", then the backing integration tests must test the "keywords" are present in the response.

**Review by nCubed — CHANGES_REQUESTED** [review]
I have left a few comments on each of the files; please update accordingly. Also, I have just committed a new .editorconfig file to the master branch which has the formatting rules you need to use to follow the same formatting used in the repo.

For the backing integration tests, it would be helpful if you could create some additional tests to validate expected data. See other areas of the integration tests for examples. Do you best; I can add more later as it can be a bit confusing.

**Comment by bariscanyilmaz** [comment]
@nCubed could you review

**Comment by nCubed** [comment]
> @nCubed could you review

Thank you for making your changes. 

1. Were you able to merge in the updated .editorcofig into your branch? Doesn't look like the formatting matches the repositories formatting.
1. Did you try to create more complete integration tests (see initial review)

I'd like to get your PR into the wrapper. We just need to ensure the code is consistent with the repositories style, formatting, and integration testing. As I mentioned in the initial review, if you are not comfortable with creating better tests, just let me know and I'll help out.

**Comment by bariscanyilmaz** [comment]
> > @nCubed could you review
> 
> Thank you for making your changes.
> 
> 1. Were you able to merge in the updated .editorcofig into your branch? Doesn't look like the formatting matches the repositories formatting.
> 2. Did you try to create more complete integration tests (see initial review)
> 
> I'd like to get your PR into the wrapper. We just need to ensure the code is consistent with the repositories style, formatting, and integration testing. As I mentioned in the initial review, if you are not comfortable with creating better tests, just let me know and I'll help out.

1- I merged your commit to this branch [here is the editor config](https://github.com/bariscanyilmaz/TheMovieDbWrapper/blob/movie-recommendation/.editorconfig)
2- Actually, help would be great because I am not experienced in tesitng.

**Comment by nCubed** [comment]
I am merging this PR into a new branch and will fix as-needed.

**Pull request merged**